src/test/java/uk/gov/pay/api/utils/mocks/CreateChargeResponseFromConnector.java

```diff
@@ -0,0 +1,176 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+public class CreateChargeResponseFromConnector {
+    public final Long amount, corporateCardSurcharge, totalAmount; 
+    public final PaymentState state; 
+    public final String chargeId, returnUrl, description, reference, email, paymentProvider, gatewayTransactionId, createdDate;
+    public final SupportedLanguage language; 
+    public final boolean delayedCapture; 
+    public final RefundSummary refundSummary; 
+    public final SettlementSummary settlementSummary; 
+    public final CardDetails cardDetails;
+    public final List<Map<?, ?>> links;
+    public final Optional<Map<String, ?>> metadata;
+
+    private CreateChargeResponseFromConnector(long amount, String chargeId, PaymentState state, String returnUrl,
+                                              String description, String reference, String email, String paymentProvider,
+                                              String gatewayTransactionId, String createdDate, SupportedLanguage language,
+                                              boolean delayedCapture, Long corporateCardSurcharge, Long totalAmount,
+                                              RefundSummary refundSummary, SettlementSummary settlementSummary,
+                                              CardDetails cardDetails, List<Map<?, ?>> links, Map<String, ?> metadata) {
+        this.amount = amount;
+        this.chargeId = chargeId;
+        this.state = state;
+        this.returnUrl = returnUrl;
+        this.description = description;
+        this.reference = reference;
+        this.email = email;
+        this.paymentProvider = paymentProvider;
+        this.gatewayTransactionId = gatewayTransactionId;
+        this.createdDate = createdDate;
+        this.language = language;
+        this.delayedCapture = delayedCapture;
+        this.corporateCardSurcharge = corporateCardSurcharge;
+        this.totalAmount = totalAmount;
+        this.refundSummary = refundSummary;
+        this.settlementSummary = settlementSummary;
+        this.cardDetails = cardDetails;
+        this.links = links;
+        this.metadata = metadata == null || metadata.isEmpty() ? Optional.empty() : Optional.of(metadata);;
+    }
+
+    public static final class CreateChargeResponseFromConnectorBuilder {
+        private Long amount, corporateCardSurcharge, totalAmount;
+        private PaymentState state;
+        private String chargeId, returnUrl, description, reference, email, paymentProvider, gatewayTransactionId, createdDate;
+        private SupportedLanguage language;
+        private Boolean delayedCapture;
+        private RefundSummary refundSummary;
+        private SettlementSummary settlementSummary;
+        private CardDetails cardDetails;
+        private List<Map<?, ?>> links = new ArrayList<>();
+        private Map<String, String> metadata = Map.of();
```

Comment:
I think should allow a Map that has values spanning `String`, `Boolean` or `Number` (I've been using `Map<String, Object>`.